src/duckling/project/map-select.component.ts

```diff
@@ -0,0 +1,91 @@
+import {Component, ViewContainerRef} from '@angular/core';
+import {MdDialog, MdDialogConfig, MdDialogRef} from '@angular/material';
+import {Observable} from 'rxjs';
+
+import {ProjectService} from './project.service';
+
+/**
+ * Dialog that allows user to choose a map. The result of the dialog is the map key the
+ * user has chosen. The Project service has the logic used to create or load the map.
+ */
+@Component({
+    selector: "dk-map-select",
+    styleUrls: ["./duckling/layout.css"],
+    template: `
+        <div *ngIf="!listLoaded">
+            <md-spinner></md-spinner>
+        </div>
+        <div *ngIf="listLoaded">
+            <dk-section headerText="Select an Existing Map">
+                <md-nav-list>
+                    <md-list-item
+                        *ngFor="let map of maps"
+                        (click)="selectMap(map)">
+                        {{map}}
+                    </md-list-item>
+                </md-nav-list>
+            </dk-section>
+            <dk-section headerText="Create a New Map">
+                <div>
+                    <dk-input class="dk-inline"
```

Comment:
I can't use dk-validated-input because I need to know the current state of the input to determine if the button is enabled.